lib/fika/types/atom.ex

```diff
@@ -0,0 +1,9 @@
+defmodule Fika.Types.Atom do
+  defstruct [:value]
+
+  defimpl String.Chars, for: Fika.Types.Atom do
+    def to_string(%{value: value}) do
+      ":#{value}"
+    end
+  end
+end
```

Comment:
I decided to keep it because it makes the usage of the atom as both values and types easier, and also avoids compiling code which would crash the beam. We could even later leverage this struct by adding a `T.Atom.new(atom_string)` which gives a meaningful message regarding this instead of the crash dump the BEAM normally gives

Reply:
What's the BEAM crash scenario you're thinking about?

Reply:
Enum.each(0..2_000_000, &:"a_&1")

Running this in Elixir shows the way converting blindly from strings ti atoms can be harmful. But this is a marginal gain